docs/DeveloperGuide.md

```diff
@@ -1,34 +1,600 @@
+        + [4.3.1. Add Appointment](#431-add-appointment)
+        + [4.3.2. Edit Appointment](#432-edit-appointment)
+        + [4.3.3. Delete Appointment](#433-delete-appointment)
+        + [4.3.4. Mark Appointment as Done](#434-mark-appointment-as-done)
+        + [4.3.5. List Appointments](#435-list-appointments)
+        + [4.3.6. Find Appointments of Patient](#436-find-appointments-of-patient)
+    * [4.4 Storage](#44-storage)
+    * [4.5 Logging](#45-logging)
+- [5. Documentation](#5-documentation)
+- [6. Testing](#6-testing)
+- [7. Useful Links](#7-useful-links)
+- [Appendices](#appendices)
+    * [Appendix A: Product Scope](#qppendix-a-product-scope)
+    * [Appendix B: User Stories](#appendix-b-user-stories)
+    * [Appendix C: Non-functional Requirements](#appendix-c-non-functional-requirements)
+    * [Appendix D: Glossary](#appendix-d-glossary)
+    * [Appendix E: Instructions for Manual Testing](#appendix-e-instructions-for-manual-testing)
 
-### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+## 1. Introduction
 
-## User Stories
+### 1.1. Purpose
+
+This document contains the architecture and software design specifications for the application HappyPills.
+
+### 1.2. Scope
+
+This guide is mainly for developers, designers and software engineers that are working on and using HappyPills.
+
+## 2. Setting Up
+
+1. Ensure that you have `Java 11` or later installed in your computer 
+2. Click [here](link to be added later?) to download the latest HappyPills JAR File
+
+## 3. Design 
+
+This section provides a high-level overview of HappyPills.
+
+### 3.1. Architecture
+
+![Architecture diagram](images/ArchitectureDiagram.PNG "Overview of the Application")
+
+The **Architecture diagram** above explains the high-level design of HappyPills.
+
+Main has one class called `HappyPills`.  
+It is responsible for:  
+- At app launch: Initializes the components in the correct sequence, and connects them up with each other.
+- At shut down: Shuts down the components.
+
+The rest of the App consists of four components.
+
+- `UI`: The UI of the App.
+
+- `Logic`: The command executor.
+
+- `Model`: Holds the data of the App in-memory.
+
+- `Storage`: Reads data from and writes data to the text file.
+
+
+The architecture of HappyPills is broken down into seven main classes:
+* `Ui`: This class handles the User Interface of the application.
+* `Parser`: This class handles the parsing and handling of user commands.
+* `Command`: This class handles all the commands the application has.
+* `Storage`: This class reads and writes data to and from text files for future use.
+* `Patient`: This class manages the data of data type Patient in memory. 
+* `Appointment`: This class manages the data of data type Appointment in memory.
+* `MedicalRecords`: This class manages the data of data type MedicalRecord in memory.
+
+### 3.2. TextUi Component
```

Comment:
Do you mean UI component of TextUi component? Might wanna revise as line 82 mentioned UI component instead